lib/src/lifecycle_module.dart

```diff
@@ -826,7 +911,12 @@ abstract class LifecycleModule extends SimpleModule with Disposable {
       _willResumeController.add(this);
       List<Future<Null>> childResumeFutures = <Future<Null>>[];
       for (var child in _childModules.toList()) {
-        childResumeFutures.add(child.resume());
+        childResumeFutures.add(new Future.sync(() {
+          child._parentContext = _activeSpan?.context;
+          return child.resume().whenComplete(() {
+            child._parentContext = null;
```

Comment:
I'm a little confused by this. Wouldn't the parent-child relationship be valid regardless of what state the modules are in? Why do we need to set and then unset the `_parentContext` reference here?

Reply:
The parent context refers to the current span. The parent's load context will be different from the parent's resume context, so if you just pass in one context and be done, you'll get every child's transition on the same parent's event.